keps/sig-node/127-usernamespaces-support/README.md

```diff
@@ -0,0 +1,852 @@
+`chown` operation over the image snapshot when it's pulled. This presents a risk
+as it potentially increases the time and the storage needed to handle the
+container images.
+
+There is not immediate mitigation for it, [we
+talked](https://lists.linuxfoundation.org/pipermail/containers/2020-September/042230.html)
+to the Linux kernel community and [they
+replied](https://lists.linuxfoundation.org/pipermail/containers/2020-September/042230.html)
+they are working on a solution for it. If the Linux kernel provides a solution
+for this problem, that would be something that container runtimes should use. It
+does not impact the kubelet nor the CRI gRPC spec.
+
+If the Linux kernel provides a solution for this problem, that would be
+something that container runtimes should use. It does not impact the kubelet nor
+the CRI gRPC spec.
+
+Another risk is exausting the disk space on the nodes if pods are repeativily
+started and stopped while using `Pod` mode. Since `Pod` mode is planned for
+phase 2 we haven't considered a mitigation for this case.
+
+#### Container Images with High IDs
+
+There are container images designed to run with high user and group IDs. It's
+possible that the IDs range assigned to the pod is not big enough to accommodate
+these IDs, in this case they will be mapped to the `nobody` user in the host.
+
+It's not a big problem in the `Cluster` case, the users have to be sure that
+they provide a range accommodating these IDs. It's more difficult to handle in
+the `Pod` case as the logic to allocate the ranges for each pod has to take this
+information in consideration. It's likely that this requires some changes to the
+CRI and kubelet so the runtimes can inform the kubelet what are the IDs present
+on a specific container image.
+
+## Implementation Phases
+
+The implemenation of this KEP in a single phase is complicated as there are many
+discussions to be done. We learned from previous attempts to bring this support in
+that it should be done in small steps to avoid losing the focus on the
+discussion. It's also true that a full plan should be agreed at the beginning to
+avoid changing the implementation drastically in further phases.
+
+This proposal implementation aims to be divided in the following phases:
+
+### Phase 1
+
+This first phase includes:
+ - Extend the PodSpec with the `userNamespaceMode` field.
+ - Extend the CRI with user and ID mappings fields.
+ - Implement support for `Host` and `Cluster` user namespace modes.
+
+The goal of this phase is to implement some initial user namespace support
+providing pod-to-host isolation and supporting volumes. The implementation of
+the `Pod` mode is out of scope in this phase because it requires a non
+negligible amount of work and we could risk losing the focus failing to deliver
+this feature.
+
+### Future Phases
+
+These phase aims to implement the `Pod` mode. After this phase is completed the
```

Comment:
Decide on singular or plural:
"These phases aim" or "This phase aims"
(ditto for section title and next sentences)